MCDataProducts/inc/CosmicLivetime.hh

```diff
@@ -0,0 +1,78 @@
+#ifndef MCDataProducts_CosmicLivetime_hh
+#define MCDataProducts_CosmicLivetime_hh
+
+// Original author Stefano Roberto Soleti
+
+// C++ includes
+#include <iostream>
+#include <vector>
+#include <math.h>
+
+// Mu2e includes
+
+namespace mu2e {
+
+  struct CosmicLivetime{
+
+  public:
+
+    CosmicLivetime():
+      _primaries(0),
+      _area(0),
+      _lowE(0),
+      _highE(0),
+      _fluxConstant(0)  {
+    }
+
+    CosmicLivetime( unsigned int primaries,
+                    float area,
+                    float lowE,
+                    float highE,
+                    float fluxConstant ):
+      _primaries(primaries),
+      _area(area),
+      _lowE(lowE),
+      _highE(highE),
+      _fluxConstant(fluxConstant) {
+        const float eslope = -2.7;
+
+        const float EiToOneMinusGamma = pow(_lowE, 1 + eslope);
+        const float EfToOneMinusGamma = pow(_highE, 1 + eslope);
+        // http://pdg.lbl.gov/2018/reviews/rpp2018-rev-cosmic-rays.pdf eq. 29.2
+        _livetime = _primaries / (M_PI * _area * _fluxConstant * (EfToOneMinusGamma - EiToOneMinusGamma) / (1. + eslope));
```

Comment:
The integral of eq 29.2 looks right to me.
However, have you already run the cross-checks with CRY and Daya Bay?

Reply:
I've normalized the events to the same cosmic live-time for CRY and CORSIKA in the plots shown in https://mu2e-docdb.fnal.gov/cgi-bin/private/RetrieveFile?docid=29724&filename=mu2emeeting20191106.pdf&version=1.

Reply:
I remember seeing your slides.
I have not realized the comparisons are based on that formula.
Looks good.